ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1703,6 +1703,126 @@ class EclCmdStatus : public EclCmdCommon
     unsigned int       optListLimit;
 };
 
+class EclCmdZapGen : public EclCmdCommon
+{
+public:
+    EclCmdZapGen()
+    {
+        optObj.accept = eclObjWuid;
+    }
+    virtual eclCmdOptionMatchIndicator parseCommandLineOptions(ArgvIterator &iter)
+    {
+        eclCmdOptionMatchIndicator retVal = EclCmdOptionNoMatch;
+        if (iter.done())
+            return EclCmdOptionNoMatch;
+
+        for (; !iter.done(); iter.next())
+        {
+            const char *arg = iter.query();
+            if (iter.matchOption(optWuid, ECLOPT_WUID)||iter.matchOption(optWuid, ECLOPT_WUID_S))
+            {
+                optObj.type = eclObjWuid;
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optPath, ECLOPT_PATH))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optIncThorSlave, ECLOPT_INC_THOR_SLAVE_LOGS))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optProblemDesc, ECLOPT_PROBLEM_DESC))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            eclCmdOptionMatchIndicator ind = EclCmdCommon::matchCommandLineOption(iter, true);
+            if (ind != EclCmdOptionMatch)
+                return ind;
+        }
+        return retVal;
+    }
+    virtual bool finalizeOptions(IProperties *globals)
+    {
+        if (!EclCmdCommon::finalizeOptions(globals))
+            return false;
+        return true;
+    }
+    virtual int processCMD()
+    {
+        if (optWuid.isEmpty())
+        {
+            fprintf(stdout, "No WUID .\n");
+            return 0;
+        }
+        Owned<IClientWsWorkunits> client = createCmdClient(WsWorkunits, *this);
+
+        Owned<IClientWUCreateZAPInfoRequest> zapReq = client->createWUCreateZAPInfoRequest();
+        zapReq->setWuid(optWuid.get());
+        zapReq->setIncludeThorSlaveLog(optIncThorSlave);
+        zapReq->setProblemDescription(optProblemDesc.get());
+
+        Owned<IClientWUCreateZAPInfoResponse> zapResp = client->WUCreateZAPInfo(zapReq);
+        int ret = outputMultiExceptionsEx(zapResp->getExceptions());
+        if (ret == 0)
+        {
+            StringBuffer filePath(optPath);
+            const char *p = filePath.str();
+            p = (p + strlen(p) - 1);
+            if (!streq(p, PATHSEPSTR))
+                filePath.append(PATHSEPSTR);
+
+            filePath.append(zapResp->getZAPFileName());
+
+            int ofile = open(filePath.str(), _O_WRONLY | _O_CREAT | _O_TRUNC, _S_IREAD | _S_IWRITE);
```

Comment:
Is there a particular reason you are not using the JLIB IFileIO routines?  For consistency.

Reply:
No, changed